### PR TITLE
Windows: Tidy/godoc volume*.go

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

This tidies daemon\volume*.go to address some Windows TODO, but in reality all it really has added is godoc. I godoc'd the _Unix file too :) No functional code changes in the end.

### CODE REVIEW

File: daemon/volumes.go [hunk 1]
```diff
@@ -59,6 +60,7 @@ func (m *mountPoint) hasResource(absolutePath string) bool {
 	return err == nil && relPath != ".." && !strings.HasPrefix(relPath, fmt.Sprintf("..%c", filepath.Separator))
 }
 
+// Path returns the path of a volume in a mount point
```
Comment (tiborvass) [inline daemon/volumes.go]: @jhowardmsft needs full stop

File: daemon/volumes_unix.go [hunk 2]
```diff
@@ -97,6 +99,7 @@ func parseBindMount(spec string, mountLabel string, config *runconfig.Config) (*
 	return bind, nil
 }
 
+// sortMounts sorts an array of Mounts
```
Comment (tiborvass) [inline daemon/volumes_unix.go]: full stop

Comment (jhowardmsft) [inline daemon/volumes_unix.go]: Grammer police :) Yeah, updating :)

**Comment by jhowardmsft** [comment]
Added a couple of full-stops.

File: daemon/volumes_windows.go [hunk 3]
```diff
@@ -7,11 +7,13 @@ import (
 	"github.com/docker/docker/runconfig"
 )
 
-// Not supported on Windows
+// copyOwnership copies the permissions and group of a source file into the
```
Comment (thaJeztah) [inline daemon/volumes_windows.go]: s/into/to/? Just a nit

**Comment by tiborvass** [comment]
@jhowardmsft i did spot them, but i assumed you got the idea ;)

**Comment by tiborvass** [comment]
LGTM Ping @cpuguy83

File: daemon/volumes.go [hunk 4]
```diff
@@ -17,8 +17,7 @@ import (
 // a volume mount that is not writable.
 var ErrVolumeReadonly = errors.New("mounted volume is marked read-only")
 
-// TODO Windows. Further platform refactoring can still be done in volumes*.go
-
+// mountPoint holds the information pertaining to a mount point.
```
Comment (cpuguy83) [inline daemon/volumes.go]: mountPoint is the intersection point between a Volume and a container. It specifies which volume is to be used and where Docker should mount it to inside a container.

Feel free to clean that up... but as is the doc is tautological.

File: daemon/volumes_unix.go [hunk 5]
```diff
@@ -97,6 +99,7 @@ func parseBindMount(spec string, mountLabel string, config *runconfig.Config) (*
 	return bind, nil
 }
 
+// sortMounts sorts an array of Mounts.
```
Comment (cpuguy83) [inline daemon/volumes_unix.go]: sortMounts sorts an array of mounts in lexicographic order. This ensures that when mounting, the mounts don't shadow other mounts, e.g. mounting `/etc` and `/etc/resolv.conf`, `/etc/resolv.conf` must not be mounted first.

File: daemon/volumes_windows.go [hunk 6]
```diff
@@ -7,11 +7,13 @@ import (
...
+// destination file. This is a no-op on Windows.
 func copyOwnership(source, destination string) error {
 	return nil
 }
 
+// setupMounts configures the mount points for a container.
```
Comment (cpuguy83) [inline daemon/volumes_windows.go]: Needs for info on what `configures` means here.

File: daemon/volumes_windows.go (hunk 3 above)
Comment (jhowardmsft) [inline daemon/volumes_windows.go]: Done. And in the _unix version.

File: daemon/volumes.go (hunk 4 above)
Comment (jhowardmsft) [inline daemon/volumes.go]: Updated (wait for push)

File: daemon/volumes_unix.go (hunk 5 above)
Comment (jhowardmsft) [inline daemon/volumes_unix.go]: Updated.

File: daemon/volumes_windows.go (hunk 3 above)
Comment (thaJeztah) [inline daemon/volumes_windows.go]: Thanks!

File: daemon/volumes_windows.go (hunk 6 above)
Comment (jhowardmsft) [inline daemon/volumes_windows.go]: Updated. And in the _unix version.

**Comment by cpuguy83** [comment]
LGTM

**Comment by thaJeztah** [comment]
:100: !! :tada: congratulations, @jhowardmsft !

Free Windows 10 upgrades for everyone?

**Pull request merged**